### PR TITLE
docs: initial docs on the design

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+but in the future, we could limit the list of components only to those
+applicable for particular cluster (right now, the analyzer exposes all the
+components the analyzer is aware of, even the ones not relevant to the cluster.)
+
+The anatomy:
+
+```
+cluster:health:components{
+    // The layer the component belongs to.
+    // Currently one of ["compute", "core", "workload"].
```
Comment (falox) [inline docs/design.md:79]: I would like to elaborate on the meaning of the 3 levels. The difference between `calculation` and `workload` may not be so clear.

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+} 50 // The ranking of the component. The more important, the lower value.
+```
+
+## cluster:health:components:map
+
+Provides mapping between:
+- source signal, e.g. alerts
+- component
+- incident
+- normalized severity
```
Comment (radekvokal) [inline docs/design.md:94]: looking at the definition bellow, is "normalized severity" an average severity from all signals or is there a different mechanism how incident gets a severity scoring?

File: docs/design.md (hunk 1 above)
Comment (iNecas) [reply docs/design.md]: Expanded

**Comment by falox** [comment]
LGTM

**Pull request merged**